main.tf

```diff
@@ -14,7 +18,7 @@ resource "aws_efs_file_system" "this" {
   throughput_mode                 = var.throughput_mode
 
   dynamic "lifecycle_policy" {
-    for_each = length(var.lifecycle_policy) > 0 ? [var.lifecycle_policy] : []
+    for_each = local.lifecycle_policy
```

Comment:
```suggestion
    for_each = length(var.lifecycle_policy) > 0 ? var.lifecycle_policy : []
```

Reply:
I had tried this case in the past against a `var.lifecycle_policy` that was a map of 0, 1, or 2 key-value pairs and could not get it to work for all three cases, so I created the extra `local.lifecycle_policy` as a shim to avoid rewriting the `content` within `dynamic "lifecycle_policy"`.

Reply:
See proposed changes here https://github.com/KAR-AUTO/terraform-aws-efs/pull/1